### ISSUE TITLE
💅  imports in vue used as type in <script> and as value in <template> are erroneously fixed

### ISSUE THREAD

**Issue by Superpat** [biomejs/biome#2239]
### Environment information

```bash
CLI:
  Version:                      1.6.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        complexity/noForEach = "off"
                                correctness/noUnusedVariables = "off"
                                style/noNonNullAssertion = "off"
                                suspicious/noConsoleLog = "off"
                                suspicious/noExplicitAny = "off"

Workspace:
  Open Documents:               0
```

### Rule name

lint/style/useImportType

### Playground link

https://codesandbox.io/p/devbox/silent-resonance-9379qh

### Expected result

When I import a component and use it in the template section, there is no linting error. If I create a ref that uses that component's type then I get a warning that it's only being used as a type and that I should add type to it's import. If I do that, the code breaks.

It should recognize that the component is being used as a value in the template part.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Superpat** [comment]
I should mention that further testing shows that this affects any import that is used a value in the template, but only as a type in the script.

**Comment by emilio-toledo** [comment]
I can also confirm that I am experiencing this and it is quite annoying, a solution would be to use the biome-ignore comment or disable that linting error altogether which is not ideal.

**Comment by ematipico** [comment]
I would suggest ignoring this specific rule for Vue files for the time being.

**Comment by nhedger** [comment]
The following override will temporarily prevent the rule from being applied/detected.

```json
"overrides": [
	{
		"include": [
			"**/*.vue"
		],
		"linter": {
			"rules": {
				"style": {
					"useImportType": "off"
				}
			}
		}
	}
],
```

**Comment by whschultz** [comment]
Biome does not parse anything in the `template`.  Therefore, any rules whose logic would be affected by something being used within the template will need to be disabled (currently?), as noted by @nhedger.

**Comment by ematipico** [comment]
Closing as it's not parsed
